server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -232,7 +235,7 @@ public static RemoteClusterStateValidationMode parseString(String mode) {
     private final String METADATA_UPDATE_LOG_STRING = "wrote metadata for [{}] indices and skipped [{}] unchanged "
         + "indices, coordination metadata updated : [{}], settings metadata updated : [{}], templates metadata "
         + "updated : [{}], custom metadata updated : [{}], indices routing updated : [{}]";
-    private final boolean isPublicationEnabled;
+    private AtomicBoolean isPublicationEnabled;
```

Comment:
We just needed a `volatile` variable here

Reply:
done